rpc/client.go

```diff
@@ -2,21 +2,84 @@ package rpc
 
 import (
 	"context"
-	"path/filepath"
+	"runtime"
+	"time"
 
 	"github.com/evergreen-ci/poplar"
 	"github.com/evergreen-ci/poplar/rpc/internal"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/queue"
 	"github.com/mongodb/grip"
 	"github.com/mongodb/grip/message"
 	"github.com/pkg/errors"
 	"google.golang.org/grpc"
 )
 
-func UploadReport(ctx context.Context, report *poplar.Report, cc *grpc.ClientConn, dryRun bool) error {
-	return errors.Wrap(uploadTests(ctx, internal.NewCedarPerformanceMetricsClient(cc), report, report.Tests, dryRun),
+type UploadReportOptions struct {
+	Report          *poplar.Report
+	ClientConn      *grpc.ClientConn
+	SerializeUpload bool
+	DryRun          bool
+}
+
+func UploadReport(ctx context.Context, opts UploadReportOptions) error {
+	if err := convertAndUploadArtifacts(ctx, opts.Report, opts.SerializeUpload, opts.DryRun); err != nil {
+		return errors.Wrap(err, "problem uploading tests for report")
+	}
+	return errors.Wrap(uploadTests(ctx, internal.NewCedarPerformanceMetricsClient(opts.ClientConn), opts.Report, opts.Report.Tests, opts.DryRun),
 		"problem uploading tests for report")
 }
 
+func convertAndUploadArtifacts(ctx context.Context, report *poplar.Report, serialize, dryRun bool) error {
```

Comment:
perhaps this should be a method on UploadReportOptions. 

also we should avoid starting the queue in serialized mode

also there should be a mechanism to shut down the queue when it's done, otherwise threads will be hanging out.